feat(workflows): support backfilling for destinations

meikelmosby opened the pull request:
## Problem

Currently we are facing the issue that Backfilling / replaying all of your messages is not possible. When you setup a destination only events from that point in time are being respected. Since folks want to backfill destinations like e.g. customer.io or similar we want to offer a solution to do that

## Changes

- adds consumer to listen to the backfill topic
- consumer has the same logic as the hog consumer as in executing destinations

## How did you test this code?

- added tests

meikelmosby committed:
feat(workflows): adds backfill consumer

meikelmosby committed:
update

Commit:
feat(workflows): set up frontend for realtime destination backfills

Commit:
tweak

Commit:
tweak

meikelmosby committed:
update

Commit:
working nicely e2e

tomasfarias committed:
chore: Add new migration

tomasfarias committed:
feat: Add new workflows destination

tomasfarias committed:
chore: Actually add the migration

meikelmosby committed:
add python topic

tomasfarias committed:
fix: Start and flush producer

meikelmosby committed:
only do backfills for destinations

rossgray committed:
Regenerate migration

tomasfarias committed:
fix: Make test pass

Commit:
tweak

tomasfarias committed:
chore: Add security_protocol fixture

tomasfarias committed:
fix: Add type hint for non_retryable_error_types

tomasfarias committed:
fix: Revert conftest.py

rossgray committed:
Add is_last_backfill_run to BackfillDetails

tomasfarias committed:
fix: Ensure ordering with key and use zstd compression

tomasfarias committed:
fix: ignore typing

rossgray committed:
Update tests

rossgray committed:
Speed up tests

tomasfarias committed:
test: Add (broken) workflow test

meikelmosby committed:
better ux

rossgray committed:
Produce final message at end of backfill

tomasfarias committed:
fix: Set default fields for workflows

tomasfarias committed:
refactor: Make key the batch export id

tomasfarias committed:
fix: Update test to use timeout

tomasfarias committed:
refactor: Use correct schema for workflows

tomasfarias committed:
fix: Parse new datetime fields

rossgray committed:
add test for final backfill run

tomasfarias committed:
fix: Schema again

tomasfarias committed:
refactor: Drop key, add batch_export_id to schema

rossgray committed:
Add todo

meikelmosby committed:
update

meikelmosby committed:
update

tomasfarias committed:
fix: Make topic not required

meikelmosby committed:
update

meikelmosby committed:
update

meikelmosby committed:
update

meikelmosby committed:
update

rossgray commented on products/batch_exports/backend/temporal/pipeline/entrypoint.py:
```diff
@@ -59,6 +59,8 @@ async def execute_batch_export_using_internal_stage(
     initial_retry_interval_seconds: int = 5,
     maximum_retry_interval_seconds: int = 120,
     override_start_to_close_timeout_seconds: int | None = None,
+    num_partitions: int | None = None,
```
just checking - did we say we're going to drop the ordering constraint?

rossgray committed:
Add migration

Make index creation concurrent

rossgray committed:
Add feature flag in API

meikelmosby committed:
remove cdp consumer logic

meikelmosby committed:
hide backfills

meikelmosby committed:
update

github-actions[bot] committed:
test(storybook): update UI snapshots

rossgray committed:
Fix test snapshots

rossgray committed:
Fix list endpoint test

rossgray committed:
Fix tests

rossgray committed:
Remove sending of final message

rossgray committed:
Remove TODO

rossgray committed:
Remove ordering from queries

rossgray committed:
Try to fix type error

rossgray committed:
Add icon for workflows batch export

meikelmosby committed:
update failing test

rossgray committed:
Separate migrations

webjunkie committed:
fix(migrations): split atomic and non-atomic operations

Move CONCURRENTLY index from 0948 to 0949 so that:
- 0948 (column add) uses atomic=True for rollback safety
- 0949 (concurrent index) uses atomic=False as required

The AlterField in 0949 is a no-op (just Django state for choices)
so it's safe to include in the non-atomic migration.

webjunkie committed:
fix(migrations): restructure for proper atomic/non-atomic separation

- Move AlterField (choices no-op) into SeparateDatabaseAndState state_operations
- Keep 0949 as purely CONCURRENTLY with atomic=False
- Fix DDL isolation check to not count nested ops inside SeparateDatabaseAndState

This follows the guideline: atomic=False only for CONCURRENTLY operations.

rossgray committed:
Fix migration reverse SQL to drop index concurrently
